component/common/net/config.go

```diff
@@ -86,29 +97,56 @@ func (c *ServerConfig) UnmarshalRiver(f func(v interface{}) error) error {
 
 // Convert converts the River-based ServerConfig into a weaveworks.Config object.
 func (c *ServerConfig) Convert() weaveworks.Config {
-	cfg := newDefaultConfig()
+	cfg := newWeaveworksDefaultConfig()
+	// use the configured http/grpc blocks, and if not, use a mixin of our defaults, and
+	// weaveworks's as a fallback
 	if c.HTTP != nil {
 		c.HTTP.Into(&cfg)
+	} else {
```

Comment:
@thampiotr had to add these since we need to cover three cases in which our settings and weaveworks are mixin-ed:
1. no configs at all, then we should use a mixin of weaveworks defaults and ours, covered in this `else` branches
2. all configs in an http block overriden, already handled
3. part of the http configs overriden, covered by using a default configuration object in `UnmarshallRiver`

Does this sound ok?

Reply:
Checked and all should be good. Also updated grpc docs to mention that by defaults a random port will be assigned